src/core/src/package_managers/YumPackageManager.py

```diff
@@ -52,6 +53,44 @@ def __init__(self, env_layer, execution_config, composite_logger, telemetry_writ
         self.yum_ps_prerequisite = 'sudo yum -y install yum-plugin-ps'
         self.yum_ps = 'sudo yum ps'
 
+        # auto OS updates
+        self.yum_cron = 'yum cron'
+        self.dnf_automatic = 'dnf-automatic'
+        self.packagekit = 'PackageKit'
+        self.auto_os_update_service_list = [self.yum_cron, self.dnf_automatic, self.packagekit]
+        self.os_patch_configuration_settings_file_path = ''
+        self.auto_update_service_enabled = False
+        self.auto_update_config_pattern_match_text = ""
+        self.download_updates_identifier_text = ""
+        self.download_updates_value = ""
+        self.apply_updates_identifier_text = ""
+        self.apply_updates_value = ""
+        self.enable_on_reboot_identifier_text = "enable_on_reboot"
+        self.enable_on_reboot_value = False
+        self.enable_on_reboot_check_cmd = ''
+
+        # commands for YUM Cron service
+        self.yum_cron_configuration_settings_file_path = '/etc/yum/yum-cron.conf'
+        self.yum_cron_install_check_cmd = 'systemctl list-unit-files --type=service | grep yum-cron.service'  # list-unit-files returns installed services, ref: https://www.freedesktop.org/software/systemd/man/systemctl.html#Unit%20File%20Commands
+        self.yum_cron_install_cmd = 'yum -y install yum-cron'
+        self.yum_cron_enable_on_reboot_check_cmd = 'systemctl is-enabled yum-cron'
+        self.yum_cron_disable_on_reboot_cmd = 'systemctl disable yum-cron'
+        self.yum_cron_config_pattern_match_text = ' = (no|yes)'
+        self.yum_cron_download_updates_identifier_text = 'download_updates'
+        self.yum_cron_apply_updates_identifier_text = 'apply_updates'
+        self.yum_cron_enable_on_reboot_identifier_text = "enable_on_reboot"
+
+        # commands for DNF Automatic updates service
+        self.dnf_automatic_configuration_file_path = '/etc/dnf/automatic.conf'
+        self.dnf_automatic_install_check_cmd = 'systemctl list-unit-files --type=service | grep dnf-automatic.service'  # list-unit-files returns installed services, ref: https://www.freedesktop.org/software/systemd/man/systemctl.html#Unit%20File%20Commands
+        self.dnf_automatic_install_cmd = 'yum -y install yum-cron'
+        self.dnf_automatic_enable_on_reboot_check_cmd = 'systemctl is-enabled yum-cron'
+        self.dnf_automatic_disable_on_reboot_cmd = 'systemctl disable yum-cron'
+        self.dnf_automatic_config_pattern_match_text = ' = (no|yes)'
+        self.dnf_automatic_download_updates_identifier_text = 'download_updates'
+        self.dnf_automatic_apply_updates_identifier_text = 'apply_updates'
+        self.dnf_automatic_enable_on_reboot_identifier_text = "enable_on_reboot"
```

Comment:
Initialize in a separate function and call from here for code organization. This code and related functions can be in a separate foldable code region for code organization. In the future, we may explore splitting up this class as it's become too long.

Reply:
Resolved